components/mediation/security/org.wso2.micro.integrator.mediation.security/src/main/java/org/wso2/micro/integrator/mediation/security/vault/EnvironmentSecretRepository.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.micro.integrator.mediation.security.vault;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.SynapseException;
+import org.wso2.securevault.DecryptionProvider;
+import org.wso2.securevault.secret.SecretRepository;
+
+import java.util.Properties;
+
+public class EnvironmentSecretRepository implements SecretRepository {
+
+    private static Log LOG = LogFactory.getLog(EnvironmentSecretRepository.class);
+
+    /* Parent secret repository */
+    private SecretRepository parentRepository;
+
+    @Override
+    public void init(Properties properties, String s) {
+        // Nothing to do here
+    }
+
+    @Override
+    public String getSecret(String alias) {
+        String rawValue = getPlainTextSecret(alias);
+        DecryptionProvider decyptProvider = CipherInitializer.getInstance().getDecryptionProvider();
+        if (decyptProvider == null) {
+            // This cannot happen unless someone mess with OSGI references
+            LOG.error("Can not proceed decryption due to the secret repository initialization error");
+            throw new SynapseException("Secret repository is null for alias " + alias);
+        }
+        return new String(decyptProvider.decrypt(rawValue.trim().getBytes()));
```

Comment:
Shall we extract this code to a common method https://github.com/wso2/micro-integrator/blob/master/components/mediation/security/org.wso2.micro.integrator.mediation.security/src/main/java/org/wso2/micro/integrator/mediation/security/vault/FileSecretRepository.java#L61.L70

Reply:
done